src/corecel/sys/Device.cc

```diff
@@ -317,6 +320,23 @@ void activate_device(Device&& device)
     CELER_DEVICE_CALL_PREFIX(SetDevice(device.device_id()));
     d = std::move(device);
 
+    // Check capability version against cmake variable; rough but better than
+    // nothing! CMake format: "70-real 72-virtual" or "35;50;72" or "
+    if (std::string(celeritas_gpu_architectures)
+            .find(std::to_string(d.capability()))
+        == std::string::npos)
```

Comment:
Just as a note, compute capabilities are forward compatible within major versions, e.g. code for 8.0 runs on device witch sm_8.6

Reply:
"Runs on" 😒